zoo/bearRouter.js

```diff
@@ -0,0 +1,92 @@
+const router = require('express').Router();
+
+const knex = require('knex');
+const knexConfig = require('../knexfile');
+
+const db = knex(knexConfig.development);
+
+
+router.get('/', async (req, res) => {
+    try{
+        const bears = await db('bears');
+        res.status(200).json(bears)
+    } catch (error) {
+        res.status(500).json({error: 'cannot find bears'})
+    }
+});
+
+router.get('/:id', async(req, res) => {
+    try{
+        const {id} = req.params
+
+        const oneBear = await db('bears')
+            .where({id})
+            .first()
+        
+        if (oneBear) {
+            const bear = await db('bears')
+            .where({id})
+            .first()
+            res.status(200).json(bear)
+        } else {
+            res.status(404).json({message: 'There is no bear here, try a different route'})
```

Comment:
You don't need lines 22-26. You just need one call to retrieve bears. If the bear doesn't exist it should return an empty array. You can then check if the array is empty before responding with an error.

Reply:
Thanks, I updated the code to reflect this. 
```suggestion
           router.get('/:id', async(req, res) => {
    try{
        const {id} = req.params
            const bear = await db('bears')
            .where({id})
            .first()
            res.status(200).json(bear)
         if (!bear) {
            res.status(404).json({message: 'There is no bear here, try a different route'})
        }
    } catch (error){
        res.status(500).json({error: 'get id error'})
    }
});
```